src/common/components/repositories-list/index.js

```diff
@@ -69,26 +70,28 @@ class RepositoriesList extends Component {
     }
   }
 
-  renderRow(snap) {
+  renderRow(snap, index) {
+    const { hasNoRegisteredNames, registerName, snapBuilds, authStore } = this.props;
     const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
 
     let latestBuild = null;
-    const snapBuilds = this.props.snapBuilds[fullName];
+    const currentSnapBuilds = snapBuilds[fullName];
 
-    if (snapBuilds && snapBuilds.success && snapBuilds.builds.length) {
-      latestBuild = snapBuilds.builds[0];
+    if (currentSnapBuilds && currentSnapBuilds.success && currentSnapBuilds.builds.length) {
+      latestBuild = currentSnapBuilds.builds[0];
     }
 
-    const registerNameStatus = this.props.registerName[fullName] || {};
+    const registerNameStatus = registerName[fullName] || {};
 
     return (
       <RepositoryRow
         key={ `repo_${fullName}` }
         snap={ snap }
         latestBuild={ latestBuild }
         fullName={ fullName }
-        authStore={ this.props.authStore }
+        authStore={ authStore }
         registerNameStatus={ registerNameStatus }
+        registerNameIsOpen={ !index && hasNoRegisteredNames }
```

Comment:
As far as I understand this will work only for first repo from the list (index = 0), right?

My understanding of requirements was that it should open first repo that doesn't have name registered. Not first repo if it doesn't have a name registered.

So if first has name registered, but second doesn't, it should open the second one. (as it will be the first on the list that doesn't have it registered)...

But maybe it's overinterpratation...